index.js

```diff
@@ -32,6 +32,38 @@ function getMainArgs() {
   return process.argv.slice(2);
 }
 
+function setOptionValue(parseOptions, option, value, result) {
+  const multiple = parseOptions.multiples &&
+    parseOptions.multiples.includes(option);
+  const withValue = parseOptions.withValue &&
+    parseOptions.withValue.includes(option);
+
+  // Normal flag: !withValue && value === undefined
+  // Normal value, withValue && value !== undefined
+  // Special case: withValue && value === undefined
+  //    store as normal for withValue with value undefined
+  // Special case: !withValue && value !== undefined
+  //    store as normal for withValue (and not a flag)
+
+  // Flags
+  // Only mark flags for plain flag without a value, expected or otherwise.
+  const isFlag = !withValue && value === undefined;
+  if (isFlag)
+    result.flags[option] = true;
```

Comment:
I hope result.flags is a null object?

Reply:
Short version: no. Hmm.

Long version

No, initialised with 
```
const result = {
    flags: {},
    values: {},
    positionals: []
  };
```

I like the purity of a null object, but it has a slightly alarming behaviour in logging for users unaware of the behaviour:

```
 % node -e 'a=Object.create(null); a.foo="bar"; console.log(a)'
[Object: null prototype] { foo: 'bar' }
```

which can lead to some support questions: https://stackoverflow.com/questions/56298481/how-to-fix-object-null-prototype-title-product/56298574

I'm ok going either way. I decided to stick with plain object for Commander in a similar situation for historical consistency and simplicity, but Commander does not accept unknown options so prototype pollution can only occur if author makes the mistake. With zero-config parsing in `parseArg` there is more reason to consider null prototype.